ivy/functional/backends/torch/experimental/linear_algebra.py

```diff
@@ -119,5 +119,17 @@ def matrix_exp(
     return torch.exp(x)
 
 
-def eig(x: torch.Tensor, /) -> Tuple[torch.Tensor]:
-    return torch.linalg.eig(x)
+def eig(x: torch.Tensor, /) -> Tuple[torch.Tensor, ...]:
+    if not torch.is_complex(x):
+        ret = torch.linalg.eig(x.to(torch.complex128))
+    else:
+        ret = torch.linalg.eig(x)
+    return tuple(ret)
```

Comment:
Was the original function implementation throwing errors before your fix? Maybe we can split this PR into two. In the first you implement eigvals and in the other one you fix eig implementation?

Reply:
Maybe this was old implementation, that is why I did so. I will try to undo the changes and check once, how it goes. Thanks